UI/Panels/Output/StepperPanel.cs

```diff
@@ -84,40 +81,72 @@ public void SetAdresses(List<ListItem> pins)
 
         private void button1_Click(object sender, EventArgs e)
         {
-            if (OnManualCalibrationTriggered != null)
-            {
-                ManualCalibrationTriggeredEventArgs eventArgs = new ManualCalibrationTriggeredEventArgs();
-                eventArgs.StepperAddress = stepperAddressesComboBox.SelectedValue.ToString();
-                eventArgs.Steps = StepValues[trackBar1.Value];
-
-                OnManualCalibrationTriggered(this, eventArgs);
-            }
+            TriggerManualCalibration();
         }
 
+        private void TriggerManualCalibration()
+        {
+            if (OnManualCalibrationTriggered == null)
+                return;
+            
+            var eventArgs = new ManualCalibrationTriggeredEventArgs
+                {
+                    StepperAddress = stepperAddressesComboBox.SelectedValue.ToString(),
+                    Steps = stepValues[trackBar1.Value]
+                };
+
+             OnManualCalibrationTriggered(this, eventArgs);
+             button2.Enabled=true;
+            
+        }
+        
         private void button2_Click(object sender, EventArgs e)
         {
-            OnSetZeroTriggered?.Invoke(stepperAddressesComboBox.SelectedValue,
-                                   new StepperConfigChangedEventArgs()
-                                   {
-                                       StepperAddress = stepperAddressesComboBox.SelectedValue.ToString()
-                                   });
+            SetZero();
         }
 
+        private void SetZero()
+        {
+            OnSetZeroTriggered?.Invoke(stepperAddressesComboBox.SelectedValue,
+                new StepperConfigChangedEventArgs()
+                {
+                    StepperAddress = stepperAddressesComboBox.SelectedValue.ToString()
+                });
+            button2.Enabled=false;
+        }
+        
+        private void enter_KeyPress(object sender, KeyPressEventArgs e)
+        {
+            switch (e.KeyChar)
+            {
+                case (char)Keys.Return:
+                case (char)Keys.D0:
+                case (char)Keys.NumPad0:
+                    SetZero();
+                    break;
+                
+                case (char)Keys.Space:
+                    TriggerManualCalibration();
+                    break;
+            }
+        }
+        
+        
         private void inputRevTextBox_Validating(object sender, CancelEventArgs e)
         {
-            if (!(sender as Control).Parent.Enabled) return;
+            if (!((Control)sender).Parent.Enabled) return;
 
-            String value = (sender as TextBox).Text.Trim();
+            var value = ((TextBox)sender).Text.Trim();
```

Comment:
We use the `object as specificType` syntax pretty much everywhere in mobiflight.
if the object is not of that type, it will be null.

You code will throw an exception. 

Please revert to old syntax.

Reply:
Thank you.